src/features/postsSlice.ts

```diff
@@ -0,0 +1,57 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { client } from '../utils/fetchClient';
+
+export interface PostsState {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+}
+
+const initialState: PostsState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+// Asynchronous Thunk for loading users
```

Comment:
The comment mentions loading users, but this thunk is actually for loading posts. Consider updating the comment to accurately reflect the functionality.